server/chat/backend/agent/orchestrator/tool_cache.py

```
@@ -0,0 +1,80 @@
+                cached = await asyncio.to_thread(client.get, key)
+                if cached is not None:
+                    _hit_counters[incident_id] = _hit_counters.get(incident_id, 0) + 1
+                    _hit_counters.move_to_end(incident_id)
+                    while len(_hit_counters) > _HIT_COUNTER_LIMIT:
+                        _hit_counters.popitem(last=False)
+                    logger.debug(
+                        "RCA tool cache HIT: incident=%s tool=%s hits=%d",
+                        incident_id, tool_name, _hit_counters[incident_id],
+                    )
+                    return json.loads(cached)
+            except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)